specs/spec_reader.go

```diff
@@ -38,7 +55,10 @@ func (r *SpecReader) loadSpecsFromFile(path string) error {
 	if err != nil {
 		return fmt.Errorf("failed to expand file variable in file %s: %w", path, err)
 	}
-	data = []byte(os.ExpandEnv(string(data)))
+	data, err = expandEnv(data)
```

Comment:
Maybe we could just use `Expand` with a custom function?
https://pkg.go.dev/os#Expand

This would remove the need for both regular expressions